compiler/main/driver.cpp

```diff
@@ -1686,6 +1694,58 @@ static void validateSettings() {
   checkUnsupportedConfigs();
 }
 
+static void maybeRelaunchInPrgEnv(int argc, char* argv[]) {
+  // Relaunch in PrgEnv-gnu if:
+  //  * platform is cray-x* or cray-ex
+  //  * CHPL_TARGET_COMPILER is llvm
+  //  * PE_ENV does not indicate PrgEnv-gnu is loaded
```

Comment:
Can you add a top-level comment about this being off by default and opt-in with `CHPL_DO_PE_ENV_CHECK`? Looking quickly I assumed this was done by default, and only knew otherwise because you told me offline.